# Higher Diploma in Science in Data Analytics
******

## Programming for Data Analytics - Project

**Author: Rodrigo De Martino Ucedo**

This Jupyter Notebook was created as part of the [Programming for Data Analytics Module](https://www.atu.ie/courses/higher-diploma-in-science-data-analytics?_gl=1%2A1bcdos0%2A_ga%2AMTE3OTU2MzQ5LjE2OTY2MDYwMzE.%2A_ga_5R02GBYV8V%2AMTcxNDMzOTE2Ni4xMS4xLjE3MTQzMzkyMDAuMC4wLjA.https://www.atu.ie/courses/higher-diploma-in-science-data-analytics?_gl=1%2A1bcdos0%2A_ga%2AMTE3OTU2MzQ5LjE2OTY2MDYwMzE.%2A_ga_5R02GBYV8V%2AMTcxNDMzOTE2Ni4xMS4xLjE3MTQzMzkyMDAuMC4wLjA.) for the course in the [Higher Diploma in Science in Data Analytics](https://www.atu.ie/courses/higher-diploma-in-science-data-analytics?_gl=1%2A1bcdos0%2A_ga%2AMTE3OTU2MzQ5LjE2OTY2MDYwMzE.%2A_ga_5R02GBYV8V%2AMTcxNDMzOTE2Ni4xMS4xLjE3MTQzMzkyMDAuMC4wLjA.) at [ATU](https://www.atu.ie/). This project is accessing the dataset [Phoenix Park Weather](https://github.com/RodrigoDMU/programming-for-data-analytics/blob/main/project/phoenixpark_weather.csv) to demonstrate the skills developed for analysis, programming and data visualization.

******

# Phoenix Park Weather

******

![PhoenixPark](phoenixparkimg.jpeg)

The analysis focuses on mean air temperature and precipitation amount recorded in Phoenix Park, Dublin, from 2004 to 2024. These two climate variables—temperature and precipitation—are key factors that affect the conditions for outdoor activities such as walking, sports, and cycling. The data provides insights into how the park's weather patterns over the past two decades might influence people's ability to engage in these activities.

### Load Data

*****

In [58]:
# Data frames. [1]
import pandas as pd
# Plotting. [2]
import matplotlib.pyplot as plt

In [59]:
# Load dataset. [3]
df = pd.read_csv("phoenixpark_weather.csv", skiprows=19)
print (df)

     year  month  meant maxtp mintp  mnmax  mnmin  rain  gmin wdsp maxgt sun
0    2004      1    5.6  13.8  -3.8    8.5    2.7  87.1  -8.9               
1    2004      2    5.1  15.1  -4.9    9.1    1.1        -9.6               
2    2004      3    6.8  16.9  -5.0   11.0    2.6  41.2  -9.2               
3    2004      4    9.1  18.8   0.4   13.2    5.0  34.6  -3.6               
4    2004      5   11.4  21.9   1.8   16.2    6.6  40.8  -2.5               
..    ...    ...    ...   ...   ...    ...    ...   ...   ...  ...   ...  ..
243  2024      7   15.3  23.2   7.9   19.7   11.0  49.9   3.1               
244  2024      8   15.8  23.9   6.7   20.7   10.9  36.8   2.2               
245  2024      9   12.8  22.6   2.2   17.0    8.5  59.2   0.2               
246  2024     10   11.7  18.7   1.2   15.1    8.3  51.1  -0.6               
247  2024     11    8.5  19.2  -2.4   11.3    5.6  57.1  -8.1               

[248 rows x 12 columns]


### Inspect Data

*****

In [60]:
# The first 5 rows of the dataset.
df.head()

year  month  meant maxtp mintp  mnmax  mnmin  rain  gmin wdsp maxgt sun
0  2004      1    5.6  13.8  -3.8    8.5    2.7  87.1  -8.9               
1  2004      2    5.1  15.1  -4.9    9.1    1.1        -9.6               
2  2004      3    6.8  16.9  -5.0   11.0    2.6  41.2  -9.2               
3  2004      4    9.1  18.8   0.4   13.2    5.0  34.6  -3.6               
4  2004      5   11.4  21.9   1.8   16.2    6.6  40.8  -2.5

In [61]:
# The last 5 rows of the dataset.
df.tail()

year  month  meant maxtp mintp  mnmax  mnmin  rain  gmin wdsp maxgt sun
243  2024      7   15.3  23.2   7.9   19.7   11.0  49.9   3.1               
244  2024      8   15.8  23.9   6.7   20.7   10.9  36.8   2.2               
245  2024      9   12.8  22.6   2.2   17.0    8.5  59.2   0.2               
246  2024     10   11.7  18.7   1.2   15.1    8.3  51.1  -0.6               
247  2024     11    8.5  19.2  -2.4   11.3    5.6  57.1  -8.1

In [62]:
# Informations of the dataset.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 248 entries, 0 to 247
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   year    248 non-null    int64  
 1   month   248 non-null    int64  
 2   meant   248 non-null    float64
 3   maxtp   248 non-null    object 
 4   mintp   248 non-null    object 
 5   mnmax   248 non-null    float64
 6   mnmin   248 non-null    float64
 7   rain    248 non-null    object 
 8   gmin    248 non-null    object 
 9   wdsp    248 non-null    object 
 10  maxgt   248 non-null    object 
 11  sun     248 non-null    object 
dtypes: float64(3), int64(2), object(7)
memory usage: 23.4+ KB


The dataset contains 248 rows and 12 columns, each representing different weather-related metrics, with some columns being properly formatted as numerical values (integers or floats) and others stored as object types.

In [70]:
# Convert object columns to numeric.
columns_to_convert = ['maxtp', 'mintp', 'rain', 'gmin', 'wdsp', 'maxgt', 'sun'] 

for col in columns_to_convert:
    df[col] = pd.to_numeric(df[col], errors='coerce')

The non-numeric values were converted in columns like `maxtp`, `mintp`, `rain`, `gmin`, `wdsp`, `maxgt`, and `sun` into numeric types.

In [64]:
# Informations of the dataset.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 248 entries, 0 to 247
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   year    248 non-null    int64  
 1   month   248 non-null    int64  
 2   meant   248 non-null    float64
 3   maxtp   247 non-null    float64
 4   mintp   247 non-null    float64
 5   mnmax   248 non-null    float64
 6   mnmin   248 non-null    float64
 7   rain    244 non-null    float64
 8   gmin    246 non-null    float64
 9   wdsp    2 non-null      float64
 10  maxgt   23 non-null     float64
 11  sun     0 non-null      float64
dtypes: float64(10), int64(2)
memory usage: 23.4 KB


In [65]:
# Count the number of null.
df.isnull().sum()

year       0
month      0
meant      0
maxtp      1
mintp      1
mnmax      0
mnmin      0
rain       4
gmin       2
wdsp     246
maxgt    225
sun      248
dtype: int64

The initial dataset had 4 missing values in the `rain` column and 0 missing values in the `meant` columin.

In [66]:
# Drop rows with missing windspeed data. [7]
df = df.dropna(subset=['rain'])

In [67]:
# Count the number of null.
df['rain'].isnull().sum()

0

After applying the `dropna()` method to remove rows with missing precipitation data, the `rain` column no longer has any missing values.
The count of missing values in the `rain` column is now 0, indicating that the cleaning operation was successful.

In [68]:
# Descibe the dataset.
df.describe()

year       month       meant       maxtp       mintp  \
count   244.000000  244.000000  244.000000  244.000000  244.000000   
mean   2014.221311    6.471311   10.200000   19.031148    0.890164   
std       5.924785    3.471696    3.968124    4.664542    4.206656   
min    2004.000000    1.000000    0.100000   10.400000  -10.800000   
25%    2009.000000    3.000000    6.600000   14.800000   -2.400000   
50%    2014.000000    6.000000    9.750000   18.600000    0.150000   
75%    2019.000000    9.250000   13.725000   23.225000    4.725000   
max    2024.000000   12.000000   17.500000   33.000000    9.500000   

            mnmax       mnmin        rain        gmin  wdsp      maxgt  sun  
count  244.000000  244.000000  244.000000  243.000000   2.0  23.000000  0.0  
mean    14.068443    6.329508   64.630328   -3.402058   0.1   3.869565  NaN  
std      4.495332    3.516901   35.665322    4.251052   0.0   2.912424  NaN  
min      3.300000   -3.100000    4.100000  -13.200000   0.1   0.000000  NaN  
25%      9.975000    3.400000   38.525000   -6.800000   0.1   0.500000  NaN  
50%     13.650000    5.700000   58.650000   -4.100000   0.1   4.000000  NaN  
75%     18.400000    9.500000   83.600000   -0.050000   0.1   6.000000  NaN  
max     23.000000   12.700000  193.900000    7.700000   0.1   8.000000  NaN

In [69]:
# Describe the informations to Mean Air Temperature (°C) and Precipitation Amount (mm).
describe = df[['meant', 'rain']].describe()
print(describe)

            meant        rain
count  244.000000  244.000000
mean    10.200000   64.630328
std      3.968124   35.665322
min      0.100000    4.100000
25%      6.600000   38.525000
50%      9.750000   58.650000
75%     13.725000   83.600000
max     17.500000  193.900000


Mean Temperature (meant):

The average temperature across all records is 10.20°C, with a minimum of 0.10°C (very cold) and a maximum of 17.50°C (very warm).
The standard deviation of 3.97°C suggests that there is moderate variation in the temperature.
Median of 9.75°C is close to the mean, indicating a relatively balanced distribution of temperatures.

Precipitation (rain):

The average precipitation is 64.63 mm, with a minimum of 4.10 mm (low rainfall) and a maximum of 193.90 mm (very high rainfall).
The standard deviation of 35.67 mm indicates significant variation in monthly precipitation amounts.
Median of 58.65 mm is below the mean, suggesting that many months have lower precipitation, but a few months with high rainfall push the average up.

### Conclusion

******

In conclusion, the analysis of mean air temperature and precipitation data from 2004 to 2024 highlights the importance of favorable weather conditions in facilitating outdoor activities in Phoenix Park. Mild temperatures and moderate precipitation create optimal conditions for walking, sports, and cycling, contributing to both individual health and the collective well-being of the community. As the climate continues to experience shifts, the park’s weather patterns will remain a key factor in determining the accessibility and enjoyment of outdoor activities. This underscores the significance of maintaining healthy, active lifestyles, which can be significantly enhanced by good weather conditions that promote outdoor engagement.

Continued efforts to adapt to climate changes, including creating sheltered areas and promoting activities suited for various weather conditions, will be vital for ensuring that Phoenix Park remains an accessible and enjoyable space for outdoor recreation for years to come.

### References

*****

### End

******

last commit in 11/01/2025